src/App.tsx

```diff
@@ -85,8 +85,14 @@ export function App() {
   // Use file code loaded from disk
   // on mount, and overwrite any locally-stored code
   useEffect(() => {
-    if (isTauri() && loadedCode !== null && loadedCode !== undefined) {
-      kclManager.setCode(loadedCode)
+    if (isTauri() && loadedCode !== null) {
+      if (kclManager.engineCommandManager.engineConnection?.isReady()) {
+        // If the engine is ready, promptly execute the loaded code
+        kclManager.setCodeAndExecute(loadedCode)
+      } else {
+        // Otherwise, just set the code and wait for the connection to complete
+        kclManager.setCode(loadedCode)
+      }
```

Comment:
I'm nervous about if this is the right way to solve this @Irev-Dev, but when I was navigating between files it wasn't re-rendering, and I suspect it was because since we were firing `setCode` it was just waiting for the stream to connect and getting the initial render automatically, but since I'd have a stream already I actually needed to request a re-render.